src/instructions.py

```diff
@@ -273,7 +290,33 @@ def initialize_market(params: InitializeMarketParams) -> TransactionInstruction:
 
 
 def new_order(params: NewOrderParams) -> TransactionInstruction:
-    raise NotImplementedError("new_order not implemented")
+    """Generate a transaction instruction to place new order."""
+    return TransactionInstruction(
+        keys=[
+            AccountMeta(pubkey=params.market, is_signer=False, is_writable=True),
+            AccountMeta(pubkey=params.open_orders, is_signer=False, is_writable=True),
+            AccountMeta(pubkey=params.request_queue, is_signer=False, is_writable=True),
+            AccountMeta(pubkey=params.payer, is_signer=False, is_writable=True),
+            AccountMeta(pubkey=params.owner, is_signer=True, is_writable=False),
+            AccountMeta(pubkey=params.base_vault, is_signer=False, is_writable=True),
+            AccountMeta(pubkey=params.quote_vault, is_signer=False, is_writable=True),
+            AccountMeta(pubkey=DEFAULT_DEX_PROGRAM_ID, is_signer=False, is_writable=False),
+            AccountMeta(pubkey=SYSVAR_RENT_PUBKEY, is_signer=False, is_writable=False),
+        ],
+        program_id=params.program_id,
+        data=INSTRUCTIONS_LAYOUT.build(
```

Comment:
What about the no `client_id` case?

Reply:
I have thought about this, but quite unsure. Because in the binary representation, it has to have a number and if the client id is `None` then how can we encode this in binary? Do we use a special value, e.g. `0`?
So right now I just assume it will always be supplied.

Reply:
Yeah. I think we can just use 0s.

Here is the behavior from serum-js:
https://repl.it/repls/RunnyPushyHertz